### PR TITLE
udev: do not link with libudev

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by bluca** [comment]
CentOS failure is https://github.com/systemd/systemd/issues/17963 and bionic-* times out in NUMA-POLICY

**Comment by bluca** [comment]
So the goal is to avoid linking a bunch of binaries with libudev, but instead to have the minimum required functions in libsystemd-shared, to reduce overall binary size, right?
The diff looks good to me, and no objections if this is the goal - let's wait a little and see if others want to chime in

File: src/udev/udev-builtin.h [hunk 1]
```diff
@@ -37,6 +37,10 @@ typedef struct UdevBuiltin {
         bool run_once;
 } UdevBuiltin;
 
+#define UDEV_NAME_SIZE   512
+#define UDEV_PATH_SIZE  1024
+#define UDEV_LINE_SIZE 16384
```
Comment (poettering) [inline src/udev/udev-builtin.h]: isn't that something to move src/shared/udev-util.h, too? It doesn't really have anything to do with builtins, no?

File: src/shared/udev-util.c [hunk 2]
```diff
@@ -536,3 +538,20 @@ int udev_resolve_subsys_kernel(const char *string, char *result, size_t maxsize,
         }
         return 0;
 }
+
+int udev_queue_is_empty(void) {
+        return access("/run/udev/queue", F_OK) < 0;
```
Comment (poettering) [inline src/shared/udev-util.c]: maybe take the opportunity here to fix error propagation on this?
i.e.

```c
int udev_queue_is_empty(void) {
        return access("/run/udev/queue") < 0 ? 
                (errno == ENOENT ? true : -errno) : false;
```

And then if the other errors really should be treated equal to "is empty" handle that in the caller?

**Review by poettering — COMMENTED** [review]
Love the clean-ups. Some minor suggestions.

**Comment by yuwata** [comment]
@bluca and @poettering Thank you for your review. I've addressed the above suggestions. Setting the green label.

> So the goal is to avoid linking a bunch of binaries with libudev, but instead to have the minimum required functions in libsystemd-shared, to reduce overall binary size, right?

Right. Though, I am not sure about the binary size change. But I hope it is reduced slightly.

**Comment by poettering** [comment]
> So the goal is to avoid linking a bunch of binaries with libudev, but instead to have the minimum required functions in libsystemd-shared, to reduce overall binary size, right?

I personally don't care too much about the binary size I must say. But I am very keen on keeping the codebase clean, and I think relegating "libudev" into "this is legacy mode", and thus not linking against it anymore from our own code is inherently a good thing.

**Pull request merged**